python/mxnet/symbol.py

```diff
@@ -1025,6 +1025,36 @@ def grad(self, wrt):
         return Symbol(handle)
     # pylint: enable= no-member
 
+    def eval(self, args=None, ctx=cpu()):
+        if args == None:
+            args = {}
+        # print("hello world!")
+        """Evaluate a symbol given arguments
+
+        The `eval` method combines a call to `bind` (which returns an executor) with a call to `forward` (executor method).
+        For the common use case where you plan to repeatedly evaluate the same symbol with the same NDArray inputs, this is slow.
+        In that case, you should call `bind` once and then repeatedly call forward. Eval allows simpler syntax for less cumbersome introspection.
+
+        Parameters
+        ----------
+        args : list of NDArray or dict of str to NDArray
+            Input arguments to the symbol.
+
+            - If type is list of NDArray, the position is in the same order of list_arguments.
+            - If type is dict of str to NDArray, then it maps the name of arguments
+              to the corresponding NDArray.
+            - In either case, all the arguments must be provided.
+
+        ctx : Context
+            The device context the generated executor to run on.
+
+        Returns
+        ----------
+        result :  a list of NDArrays corresponding to the values taken by each symbol when evaluated on given args. When called on a single symbol (not a group), the result will be a list with one element.
+        """
+        return self.bind(ctx, args).forward()
```

Comment:
I thought about this but decided against it. The main reason is I think it should return the exact same thing as `bind` + `forward`. If in one case it returns a list but in the other case it returns an NDArray then that makes for a confusing interface.